### ISSUE TITLE
Distinction between Providers and Consumers

### ISSUE THREAD

**Issue by elballa** [siom79/japicmp#234]
Hi,
has japicmp plugin a way to distinguish between interfaces designed for providers (like javax.servlet.ServletContext) and those designed for consumers (like javax.servlet.Servlet) as bnd-baseline does?
Provider is a person implementing an API and a Consumer is a person using the API.
Marking interface for a specific "person" can change the severty of the source code changes in terms of backward compatibility.
In the examples of the Servlet API, adding a method to the ServletContext would be a minor change because it only requires an update to the providers (like Tomcat or Jetty) and has no any impact to the consumers. Adding a method to the Servlet interface, however, would break all WAR files and therefore be a major change.
Thanks

**Comment by siom79** [comment]
No, japicmp does not distinguish between provider and consumer.

Am Do., 11. Apr. 2019, 10:24 hat elballa <notifications@github.com>
geschrieben:

**Comment by elballa** [comment]
Hi,
do you plan to provide this new feature in a future release?
Thanks

**Comment by siom79** [comment]
When I find some time, I will tackle that.
